pilot/pkg/networking/core/v1alpha3/httproute.go

```diff
@@ -290,6 +295,61 @@ func selectVirtualServices(virtualServices []config.Config, servicesByName map[h
 	return out
 }
 
+type ProxyHeaders struct {
+	ServerName                 string
+	ServerHeaderTransformation hcm.HttpConnectionManager_ServerHeaderTransformation
+	ForwardedClientCert        hcm.HttpConnectionManager_ForwardClientCertDetails
+	IncludeRequestAttemptCount bool
+	GenerateRequestID          *wrappers.BoolValue
+	SuppressDebugHeaders       bool
+}
+
+func GetProxyHeaders(node *model.Proxy, push *model.PushContext, class istionetworking.ListenerClass) ProxyHeaders {
+	pc := node.Metadata.ProxyConfigOrDefault(push.Mesh.DefaultConfig)
+	return GetProxyHeadersFromProxyConfig(pc, class)
+}
+
+func GetProxyHeadersFromProxyConfig(pc *meshconfig.ProxyConfig, class istionetworking.ListenerClass) ProxyHeaders {
+	base := ProxyHeaders{
+		ServerName:                 EnvoyServerName,
+		ServerHeaderTransformation: hcm.HttpConnectionManager_OVERWRITE,
+		ForwardedClientCert:        hcm.HttpConnectionManager_APPEND_FORWARD,
+		IncludeRequestAttemptCount: true,
+		SuppressDebugHeaders:       false,
+		GenerateRequestID:          nil, // Envoy default is to enable them, so set nil
```

Comment:
move seting the default values to Mesh.DefaultConfig

Reply:
These are not the default Mesh config values, but the transformed values. So I am not sure what putting a default in mesh.DefaultConfig adds. In fact, the default mesh config is `proxyHeaders` unset.

This `ProxyHeaders` field is a transformation of mesh config values into values that can directly be used in XDS gen. For example, `ForwardedClientCert` is an Envoy type instead of an Istio type. We have `SuppressDebugHeaders` vs `proxyHeaders.EnvoyDebugHeaders.Disabled=true`.